### DISCUSSION TITLE
Unable to use pydantic validators on Relation attributes

### DISCUSSION THREAD

**Discussion by CisterMoke** [fastapi/sqlmodel#1927] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import Field, Relationship, SQLModel

class Address(SQLModel, table=True):
    id: Optional[int] = Field(primary_key=True)
    country: str
    city: str
    street: str
    street_number: int

class Building(SQLModel, table=True):
    id: int = Field(primary_key=True)
    address_id: int = Field(foreign_key=f"{ADDRESSES}.id")
    address: Address = Relationship()
    age: int

    class Config:
        arbitrary_types_allowed = True

    @validator('address', pre=True)
    def validate_address(cls, v):
        return Address.parse_obj(v)

# Importing Building produces the error "pydantic.errors.ConfigError: Validators defined with incorrect fields: validate_address (use check_fields=False if you're inheriting from the model and intended this)"
```

### Description

I want to parse a `Building` instance from a nested dictionary `d` with the `Building` and `Address` class defined in the example code.
 ```
d = {
   'address': {
      'country': 'US',
      'city': 'New York',
      'street': 'Centre St',
      'street_num': 1},
    'age': 40
}
```
So far I have been unavailable to successfully execute  `Building.parse_obj(d)`. Commenting out the validator in `Building` solves the pydantic error bein raised but the address of the parsed building will be a dict instead of an `Address` instance

### Operating System

Windows

### SQLModel Version

0.0.4

### Python Version

3.6.12

**Comment by Sillocan** [comment]
Did you ever figure out a solution?

**Comment by Raphasha27** [comment]
This happens because SQLModel replaces the field with a SQLAlchemy relationship descriptor on the table model, so Pydantic validators see the relationship object instead of raw data.

**Workaround 1: use `model_validator` (Pydantic v2)**

```python
from pydantic import model_validator

class TeamMember(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    name: str
    team_id: int | None = Field(default=None, foreign_key="team.id")
    team: "Team" = Relationship(back_populates="members")

    @model_validator(mode="before")
    @classmethod
    def validate_team_name(cls, data: Any) -> Any:
        if isinstance(data, dict) and "team" in data and isinstance(data["team"], dict):
            team_name = data["team"].get("name", "")
            if len(team_name) < 3:
                raise ValueError("team name must be at least 3 characters")
        return data
```

**Workaround 2: use a separate Pydantic model for input**

Separate your API schema from your DB model:

```python
class TeamMemberCreate(SQLModel):
    name: str
    team: TeamCreate  # Pydantic model with validators

class TeamMember(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    name: str
    team_id: int | None = Field(default=None, foreign_key="team.id")
    team: "Team" = Relationship(back_populates="members")
```

Then validate with `TeamMemberCreate` in your endpoint and create the DB model separately. This is the recommended pattern for complex validation in FastAPI + SQLModel.
